examples/hellogrpc/e2e-test.sh

```diff
@@ -18,156 +16,189 @@ set -e
+    local message="${3:-Expected '$substring' not found in '$complete'}"
 
-  echo Checking "$1" contains "$2"
-  CONTAINS "${complete}" "${substring}" || fail "$message"
+    CONTAINS "$complete" "$substring" || fail "$message"
 }
 
 function CONTAINS_PATTERN() {
-  local complete="${1}"
-  local substring="${2}"
+    local complete=$1
+    local substring=$2
 
-  echo "${complete}" | grep -Esq -- "${substring}"
+    echo "$complete" | grep -Esq -- "$substring" >/dev/null
 }
 
 function EXPECT_CONTAINS_PATTERN() {
-  local complete="${1}"
-  local substring="${2}"
-  local message="${3:-Expected '${substring}' not found in '${complete}'}"
+    local complete=$1
+    local substring=$2
+    local message="${3:-Expected '$substring' not found in '$complete'}"
 
-  echo Checking "$1" contains pattern "$2"
-  CONTAINS_PATTERN "${complete}" "${substring}" || fail "$message"
+    CONTAINS_PATTERN "$complete" "$substring" || fail "$message"
 }
 
 # Ensure there is an ip address for hell-grpc-staging:50051
 apply-lb() {
-  echo Applying service...
-  bazel build $BAZEL_FLAGS examples/hellogrpc:staging-service.apply
-  bazel-bin/examples/hellogrpc/staging-service.apply
+    logfail bazel build examples/hellogrpc:staging-service.apply
+    logfail bazel-bin/examples/hellogrpc/staging-service.apply
 }
 
 create() {
-  echo Creating $LANGUAGE...
-  bazel build $BAZEL_FLAGS examples/hellogrpc/${LANGUAGE}/server:staging.create
-  bazel-bin/examples/hellogrpc/${LANGUAGE}/server/staging.create
+    logfail bazel build "examples/hellogrpc/$1/server:staging.create"
+    logfail "bazel-bin/examples/hellogrpc/$1/server/staging.create"
 }
 
 check_msg() {
-  bazel build $BAZEL_FLAGS examples/hellogrpc/${LANGUAGE}/client
-
-  while [[ -z $(get_lb_ip $local hello-grpc-staging) ]]; do
-    echo "service has not yet received an IP address, sleeping for 5s..."
-    sleep 10
-  done
-
-  # Wait some more for after IP address allocation for the service to come
-  # alive
-  echo "Got IP Adress! Sleeping 30s more for service to come alive..."
-  sleep 30
-
-  # Make Bazel generate a temporary script that runs the client executable.
-  tmp_exec=hellogrpc_${LANGUAGE}_client
-  # This will only generate the temp executable. It won't actually run it.
-  bazel run $BAZEL_FLAGS //examples/hellogrpc/${LANGUAGE}/client --script_path=${tmp_exec}
-  OUTPUT=$(./${tmp_exec} $(get_lb_ip $local hello-grpc-staging))
-  rm ${tmp_exec}
-  echo Checking response from service: "${OUTPUT}" matches: "DEMO$1<space>"
-  echo "${OUTPUT}" | grep "DEMO$1[ ]"
+    logfail bazel build "examples/hellogrpc/$1/client"
+
+    echo -n "IP: "
+    local ip
+    while true; do
+        ip=$(get_lb_ip $local hello-grpc-staging)
+        if [[ -n "$ip" ]]; then
+            echo "$ip"
+            break
+        fi
+        echo -n .
+        sleep 5
+    done
+
+    local output
+    local tmp_exec=hellogrpc_$1_client
+    logfail bazel run "//examples/hellogrpc/$1/client" "--script_path=$tmp_exec"
```

Comment:
```suggestion
    # Make Bazel generate a temporary script that runs the client executable
    # This will only generate the temp executable. It won't actually run it.
    local output
    local tmp_exec=hellogrpc_$1_client
    logfail bazel run "//examples/hellogrpc/$1/client" "--script_path=$tmp_exec"
```